### PR TITLE
HCK-3566: Change in comments and constraints to the SQL Server target

### PR DESCRIPTION
**Pull request by oleksandr-lytovchenko**
List of implemented:

- Refactor alter scripts using DTO
- Added support for check constraints in alter scripts
- Added support for not null constraints in alter scripts
- Added support for primary keys in alter scripts
- Added support for foreign keys in alter scripts

### CODE REVIEW

File: forward_engineering/api.js [hunk 1]
```diff
@@ -4,25 +4,102 @@ const logInfo = require('../reverse_engineering/helpers/logInfo');
...
+/**
+ * @typedef {[ContainerJsonSchema, ContainerStyles]} ContainerData
+ * */
+/**
+ * @typedef {{
+ *     [id: string]: EntityJsonSchema
+ * }} EntitiesJsonSchema
+ */
+
+const parseEntities = (entities, serializedItems) => {
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/api.js:44]: Move parsing functions to script builder

File: forward_engineering/helpers/alterScriptHelpers/alterColumnHelpers.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+module.exports = (app, ddlProvider) => {
+	const _ = app.require('lodash');
+	const { checkFieldPropertiesChanged, modifyGroupItems, setIndexKeys } = require('./common')(app);
+	const { createColumnDefinitionBySchema } = require('./createColumnDefinition')(_);
+	const { AlterScriptDto } = require('./types/AlterScriptDto');
+
+	const getRenameColumnScripts = (collectionProperties, fullName) => {
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/helpers/alterScriptHelpers/alterColumnHelpers.js:9]: Move getRenameColumnScripts & getChangeTypeScripts to own files (into alterColumnHelpers folder). Use PostgreSQL for reference

File: forward_engineering/helpers/alterScriptHelpers/alterEntityHelper.js [hunk 3]
```diff
@@ -1,13 +1,35 @@
+/**
+ * @typedef {Object} Collection
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/helpers/alterScriptHelpers/alterEntityHelper.js:2]: Better to extract complex types into own js files into "types" folder - see postgres for reference

File: forward_engineering/helpers/alterScriptHelpers/alterEntityHelper.js [hunk 4]
```diff
@@ -45,20 +67,29 @@ module.exports = (app, options) => {
 			jsonSchema,
 			idToNameHashTable,
 		});
-		const tableScript = ddlProvider.createTable(hydratedTable, jsonSchema.isActivated);
+		const tableScriptDto = AlterScriptDto.getInstance([ddlProvider.createTable(hydratedTable, jsonSchema.isActivated)], true, false);
+		const indexesScriptsDto = AlterScriptDto.getInstances(indexesScripts, true, false);
 
-		return [tableScript, ...indexesScripts].join('\n\n');
+		return [tableScriptDto, ...indexesScriptsDto];
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/helpers/alterScriptHelpers/alterEntityHelper.js:73]: Just to be safe, when you return an array of AlterScriptDTOs, add ".filter(Boolean)" at the end, as factory methods may return undefined

File: forward_engineering/utils/generalUtils.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+	return chain;
+};
+
+const isEscaped = (name) => /`[\s\S]*`/.test(name);
+
+const prepareName = (name = '') => {
+	const containSpaces = /[\s-]/g;
+	if (containSpaces.test(name) && !isEscaped(name)) {
+		return `\`${name}\``;
+	} else if (RESERVED_WORDS_AS_ARRAY.includes(name.toLowerCase())) {
```
Comment (mtseluiko) [inline forward_engineering/utils/generalUtils.js:57]: but reversed words array is not in lowercase!

File: forward_engineering/enums/reservedWords.js [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+	ASSERTION: 'ASSERTION',
+	ALTER: 'ALTER',
+	AUTHORIZATION: 'AUTHORIZATION',
+	'ALTER TABLE': 'ALTER TABLE',
+	AUTOINCREMENT: 'AUTOINCREMENT',
+	And: 'And',
+	Avg: 'Avg',
+	BEGIN: 'BEGIN',
+	COLLATION: 'COLLATION',
+	Between: 'Between',
```
Comment (mtseluiko) [inline forward_engineering/enums/reservedWords.js:20]: check upper cases

**Review by mtseluiko — APPROVED** [review]

File: forward_engineering/enums/reservedWords.js [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+	SECOND: 'SECOND',
+	STDEV: 'STDEV',
+	SELECT: 'SELECT',
+	STDEVP: 'STDEVP',
+	SELECTSCHEMA: 'SELECTSCHEMA',
+	STRING: 'STRING',
+	SELECTSECURITY: 'SELECTSECURITY',
+	SUBSTRING: 'SUBSTRING',
+	SET: 'SET',
+	Sum: 'Sum',
```
Comment (mtseluiko) [inline forward_engineering/enums/reservedWords.js:200]: here still need to be uppercase

**Pull request merged**